### PR TITLE
Add "get_repo_permission" to Team class (for #1415)

### PR DESCRIPTION
**Pull request by tl-adrian-bridgett**
This adds support in a way that matches GitHub API:
https://developer.github.com/v3/teams/#check-if-a-team-manages-a-repository

I've tried to stick to the coding standards but I'm not _too_ convinced.

BTW this account is going to disappear on Friday (change of job) and/or I may well stop being able to respond :-(

### CODE REVIEW

**Comment by s-t-e-v-e-n-k** [comment]
I sort of envisioned it checking the status code returned by GitHub and returning a boolean, True if the team has permissions and False if not. There's a tox environment to run the formatting to help with that: ``tox -eformat``

WRT your account, that's disappointing, but Github accounts do support multiple email addresses

**Comment by tl-adrian-bridgett** [comment]
Ah. I'm really after the full set of permissions (admin/push/pull etc) rather than just the boolean value. Hmm. Maybe it should be returning the Permission class though if possible.

**Comment by s-t-e-v-e-n-k** [comment]
That sounds like a much better fit, indeed. Also, please add at least one test -- preferably two, to check the you don't have any permissions on this repo case. You could also probably extend the method to take a repo or a string as the repo name, but I'm not wedded to that idea.

**Comment by tl-adrian-bridgett** [comment]
Good news and bad news
Good news - I've made the improvements (and added my personal email address to github)
Bad news - I can't get the tests to pass but not sure what's going on!

**Comment by s-t-e-v-e-n-k** [comment]
This is going to be a fun rebase. :-P

The test failures are because you're making a new API call in the tests, and the replay data (tests/ReplayData/Team.testRepos.txt in this case) doesn't know about that the call, and so is matching with a different call.

If you get stuck, mention me, and I'll dig in further.

**Comment by s-t-e-v-e-n-k** [comment]
Must remember to put "Fixes #1415" in your commit body when I merge it...

**Comment by tl-adrian-bridgett** [comment]
Should be good now, I updated the replay data by using a temporary personal repo then merge/update the two new requests.  Thanks so much for all your help, I'd never have made it without you :-)

File: tests/Team.py [hunk 1]
```diff
@@ -114,11 +116,15 @@ def testRepos(self):
...
         self.assertFalse(self.team.has_in_repos(repo))
+        self.assertIsNone(self.team.get_repo_permission(repo))
         self.team.add_to_repos(repo)
         self.assertListKeyEqual(
             self.team.get_repos(), lambda r: r.name, ["FatherBeaver"]
         )
         self.assertTrue(self.team.has_in_repos(repo))
+        self.assertIsInstance(
+            self.team.get_repo_permission(repo), github.Permissions.Permissions
+        )
```
Comment (s-t-e-v-e-n-k) [inline tests/Team.py]: I'd rather you assert something in the returned object, such as:

```
permissions = self.team.get_repo_permission(repo)
self.assertTrue(permissions.pull)
```

Comment (tl-adrian-bridgett) [reply tests/Team.py]: Sounds good, that also drops the need for the import.  I've force-pushed (first time I used github UI to resolve the Const conflict but now I've rebased and fixed it there instead)

**Pull request merged**